geth/api/backend.go

```diff
@@ -224,3 +225,12 @@ func (m *StatusBackend) registerHandlers() error {
 	rpcClient.RegisterHandler("eth_sendTransaction", m.txQueueManager.SendTransactionRPCHandler)
 	return nil
 }
+
+// ConnectionChange handles network state changes logic.
+func (m *StatusBackend) ConnectionChange(state ConnectionState) {
+	log.Info("Network state change", "old", m.connectionState, "new", state)
+	m.connectionState = state
```

Comment:
could it be that events will be processed concurrently? for example, if network state flips all the time

Reply:
No idea. Theoretically yes and it's trivial to guard with mutex, but that's rather POC than actual implementation. I plan to monitor log messages and play with it on a real device as soon as `status-react` part is implemented.